src/iam.ts

```diff
@@ -1493,4 +1494,72 @@ export class IAM extends IAMBase {
       })
     );
   }
+
+  // ### ASSETS ###
+  public async registerAsset() {
+    if (!this._address) {
+      throw new Error(ERROR_MESSAGES.USER_NOT_LOGGED_IN);
+    }
+    const asset = new OfferableIdentityFactory();
+    await asset.deploy(this._address, this._assetManagerAddress);
+  }
+
+  public async offerAsset({ assetDID, offerTo }: { assetDID: string; offerTo: string }) {
```

Comment:
@JGiter It looks like the `OfferableIdentityOperator` isn't being used here: https://github.com/energywebfoundation/ew-did-registry/blob/development/packages/proxyIdentity/src/offerableIdentityOperator.ts 
Does that make sense? Or should the offerableIdentityOperator be used?

Reply:
I think that Operator/Resolver purpose is resolving DID document. It is possible to include `offer` to `OfferableIdentityOperator`, but asset contract still has to be invoked directly since `accept` and `reject` are not applicable to Operator